models/login_source.go

```diff
@@ -696,15 +698,26 @@ func LoginViaPAM(user *User, login, password string, sourceID int64, cfg *PAMCon
 
 	// Allow PAM sources with `@` in their name, like from Active Directory
 	username := pamLogin
+	email := pamLogin
 	idx := strings.Index(pamLogin, "@")
 	if idx > -1 {
 		username = pamLogin[:idx]
 	}
+	if ValidateEmail(email) != nil {
+		if cfg.EmailDomain != "" {
+			email = fmt.Sprintf("%s@%s", pamLogin, cfg.EmailDomain)
+		} else {
+			email = fmt.Sprintf("%s@%s", pamLogin, setting.Service.NoReplyAddress)
```

Comment:
Is `pamLogin` a good choice for the email? Line 699 states it can contain `@`. `username` contains the stripped name, so maybe `{username}@...` is better?